packages/react/src/components/UIShell/SideNav.tsx

```diff
@@ -101,7 +103,7 @@ function SideNavRenderFunction(
   //   : t('carbon.sidenav.state.closed');
 
   const className = cx(customClassName, {
-    [`${prefix}--side-nav`]: true,
+    [`${prefix}--side-nav`]: false,
```

Comment:
I don't think we want to set this class to false, this is where all of the style regressions are coming from. If we need to modify the `isRail` styles, we should add a new class and unset / reset styles on an as-needed basis.

Reply:
@tw15egan : Yeah correct !  I will make the change and push the code.

Reply:
@tw15egan : While working on this , I got one more thought, what if,   isRail and enterDelay is enabled to hide the expansion for a few seconds, but  user is intentionally clicking the button expecting to see it immediately. 

I just introduced a onClick event to override the timedelay set by the developer, if user wants to see it immediately ! .  Let me know if this is not needed